src/controlplane/client/authenticator/authenticator.go

```diff
@@ -115,6 +116,29 @@ type certificatesData struct {
 	ca   []byte
 }
 
+// certificateSecretKeys contains the name of the keys inside the secret where the certificate, private key, and CA
+// certificates are stored.
+type certificateSecretKeys struct {
+	cert string
+	key  string
+	ca   string
+}
+
+// opaqueCertKey is the key for the secret data where the PEM-encoded certificate is located.
+const opaqueCertKey = "cert"
+
+// opaqueKeyKey is the key for the secret data where the PEM-encoded private key is located.
```

Comment:
do we want to say somehow that this is legacy? (if we want that of course)

Reply:
agree, perhaps a general comment to the `certificateSecretKeys` struct mentioning the legacy NR naming for this secrets keys.